File: .API/Cloud/Friend.cs
// Decompiled with JetBrains decompiler
// Type: CloudX.Shared.Friend
// Assembly: CloudX.Shared, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 6223B97A-06A5-46CB-9E10-78604961D6EE
// Assembly location: J:\D\SteamLibrary\steamapps\common\NeosVR\HeadlessClient\CloudX.Shared.dll

using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System;
using System.Text.Json.Serialization;

namespace CloudX.Shared
{
  [JsonObject(MemberSerialization = MemberSerialization.OptIn)]
  public class Friend
  {
    [JsonProperty(PropertyName = "id")]
    [JsonPropertyName("id")]
    public string FriendUserId { get; set; }

    [JsonProperty(PropertyName = "ownerId")]
    [JsonPropertyName("ownerId")]
    public string OwnerId { get; set; }

    [JsonProperty(PropertyName = "friendUsername")]
    [JsonPropertyName("friendUsername")]
    public string FriendUsername { get; set; }

    [JsonProperty(PropertyName = "friendStatus")]
    [JsonPropertyName("friendStatus")]
    [Newtonsoft.Json.JsonConverter(typeof (StringEnumConverter))]
    [System.Text.Json.Serialization.JsonConverter(typeof (JsonStringEnumConverter))]
    public FriendStatus FriendStatus { get; set; }

    [JsonProperty(PropertyName = "isAccepted")]
    [JsonPropertyName("isAccepted")]
    public bool IsAccepted { get; set; }

    [JsonProperty(PropertyName = "userStatus")]
    [JsonPropertyName("userStatus")]
    public UserStatus UserStatus { get; set; }

    [JsonProperty(PropertyName = "latestMessageTime")]
    [JsonPropertyName("latestMessageTime")]
    public DateTime LatestMessageTime { get; set; }

    [JsonProperty(PropertyName = "profile")]
    [JsonPropertyName("profile")]
    public UserProfile Profile { get; set; }
  }
}

File: API/Friend.js
const { UserStatus } = require("./UserStatus");
const { UserProfile } = require("./UserProfile");
class Friend {
	/**
   *Creates an instance of Friend.
   * @param {*} $b
   * @memberof Friend
   */
	constructor($b) {
		if (!$b) $b = {};
		this.FriendUserId = $b.id;
		this.OwnerId = $b.ownerId;
		this.FriendUsername = $b.friendUsername;
		this.FriendStatus = $b.friendStatus;
		this.IsAccepted = $b.isAccepted;
		this.UserStatus = new UserStatus($b.userStatus);
		this.LatestMessageTime = $b.latestMessageTime;
		this.Profile = new UserProfile($b.profile);
	}
	/**
   *
   *
   * @param {Friend} other
   * @memberof Friend
   */
	IsSame(other) {
		if (
			this.FriendUserId === other.FriendUserId &&
      this.OwnerId === other.OwnerId &&
      this.FriendUsername === other.FriendUsername &&
      this.IsAccepted === other.IsAccepted &&
      this.FriendStatus === other.FriendStatus &&
      this.LatestMessageTime === other.LatestMessageTime &&
      this.UserStatus.IsSame(other.UserStatus)
		)
			return true;
		return false;
	}
}
module.exports = {
	Friend,
};
